File: 1288-remove-covered-intervals/1288-remove-covered-intervals.java
class Solution {
    public int removeCoveredIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> {
                   if (a[0] == b[0]) {
                       return b[1] - a[1];
                   } else {
                       return a[0] - b[0];
                   }
        });
        
        int count = 0;
        int left = intervals[0][0];
        int right = intervals[0][1];
        for (int i = 1; i < intervals.length; i++) {
            if (intervals[i][1] <= right) {
                count++;
            } else {
                left = intervals[i][0];
                right = intervals[i][1];
            }
        }
        
        return intervals.length - count;
    }
}

File: 1288-remove-covered-intervals/1288-remove-covered-intervals.py
class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        count = len(intervals)
        intervals.sort(key=lambda x: (x[0], -x[1]))
        
        for i in range(1, len(intervals)):
            if intervals[i][0] >= intervals[i - 1][0] and intervals[i][1] <= intervals[i - 1][1]:
                count -= 1
                intervals[i][0] = intervals[i - 1][0]
                intervals[i][1] = intervals[i - 1][1]
        
        return count